brillig_vm/src/lib.rs

```diff
@@ -205,32 +205,46 @@ impl VM {
 
                 for (destination, values) in destinations.iter().zip(values) {
                     match destination {
-                        RegisterValueOrArray::RegisterIndex(index) => {
+                        RegisterOrMemory::RegisterIndex(value_index) => {
                             assert_eq!(
                                 values.len(),
                                 1,
-                                "Function result size does not match brillig bytecode"
+                                "Function result size does not match brillig (expected 1 result)"
```

Comment:
This is a nit but we should settle on when we refer to just "Brillig" if we mean the VM or the bytecode. I lean towards that Brillig is the VM interface. I think here "brillig bytecode" is a bit clearer as it says the we do not match the specification in the ForeignCall opcode. Same for the other asserts in this loop.